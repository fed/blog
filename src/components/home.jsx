import React, { Fragment, useRef } from 'react';
import styled from 'styled-components';

import { socialLinks } from '../data';
import { fontFamilySansSerif, sizeContainerLarge, sizeContainerExtraLarge } from '../styles/mixins';
import { Footer } from './footer';
import { Hero } from './hero';
import { Social } from './social';

const Container = styled.div`
    padding: 32px 0;
    @media (min-width: ${sizeContainerLarge}) {
        align-content: space-between;
        display: flex;
        padding: 64px;
    }
    @media (min-width: ${sizeContainerExtraLarge}) {
        margin: 0 auto;
        max-width: 1500px;
        padding: 64px 92px;
    }
`;

const Header = styled.header`
    @media (min-width: ${sizeContainerLarge}) {
        padding: 0 30px 0 0;
        width: 40%;
    }
    @media (min-width: ${sizeContainerExtraLarge}) {
        width: 30%;
    }
`;

const Content = styled.main`
    @media (min-width: ${sizeContainerLarge}) {
        display: flex;
        justify-content: flex-end;
        padding: 0 0 0 30px;
        width: 60%;
    }

    @media (min-width: ${sizeContainerExtraLarge}) {
        width: 70%;
    }
`;

const SkipToMainContentButton = styled.button`
    font-family: ${fontFamilySansSerif};
    background-color: #dc2a5f;
    border: 1px solid #ccc;
    color: #fff;
    height: 50px;
    left: 0;
    position: fixed;
    top: 0;
    transform: translateY(-50px);
    :focus {
        transform: translateY(0);
    }
`;

export function Home(props) {
    const mainContent = useRef(null);
    const handleButtonClick = () => {
        mainContent.current.querySelector('a').focus();
    };

    return (
        <Fragment>
            <SkipToMainContentButton type="button" onClick={handleButtonClick}>
                Skip to main content
            </SkipToMainContentButton>
            <Container>
                <Header>
                    <Hero />
                    <Social links={socialLinks} />
                </Header>
                <Content ref={mainContent} aria-label="Blog posts">
                    {props.children}
                </Content>
            </Container>
            <Footer links={socialLinks} />
        </Fragment>
    );
}
