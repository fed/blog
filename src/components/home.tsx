import React, { useRef } from 'react';
import styled from 'styled-components';

import { fontFamilySansSerif, sizeContainerLarge, sizeContainerExtraLarge } from '../styles/mixins';
import { Footer } from './footer';
import { Hero } from './hero';
import { Social } from './social';

const Container = styled.div`
    padding: 32px 0;
    @media (min-width: ${sizeContainerLarge}) {
        align-content: space-between;
        display: flex;
        padding: 64px;
    }
    @media (min-width: ${sizeContainerExtraLarge}) {
        margin: 0 auto;
        max-width: 1500px;
        padding: 64px 92px;
    }
`;

const Header = styled.header`
    @media (min-width: ${sizeContainerLarge}) {
        padding: 0 30px 0 0;
        width: 40%;
    }
    @media (min-width: ${sizeContainerExtraLarge}) {
        width: 30%;
    }
`;

const Content = styled.main`
    @media (min-width: ${sizeContainerLarge}) {
        display: flex;
        justify-content: flex-end;
        padding: 0 0 0 30px;
        width: 60%;
    }

    @media (min-width: ${sizeContainerExtraLarge}) {
        width: 70%;
    }
`;

const SkipToMainContentButton = styled.button`
    font-family: ${fontFamilySansSerif};
    background-color: #dc2a5f;
    border: 1px solid #ccc;
    color: #fff;
    height: 50px;
    left: 0;
    position: fixed;
    top: 0;
    transform: translateY(-50px);
    :focus {
        transform: translateY(0);
    }
`;

interface Props {
    children?: React.ReactNode;
}

export const Home: React.FC<Props> = ({ children }) => {
    const mainContent = useRef(null);
    const handleButtonClick = () => {
        mainContent.current.querySelector('a').focus();
    };

    return (
        <>
            <SkipToMainContentButton type="button" onClick={handleButtonClick}>
                Skip to main content
            </SkipToMainContentButton>
            <Container>
                <Header>
                    <Hero />
                    <Social />
                </Header>
                <Content ref={mainContent} aria-label="Blog posts">
                    {children}
                </Content>
            </Container>
            <Footer />
        </>
    );
};
