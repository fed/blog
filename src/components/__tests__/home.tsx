import React from 'react';
import renderer from 'react-test-renderer';

import { Home } from '../home';

describe('Home', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Home>
                    <p>Content</p>
                </Home>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
