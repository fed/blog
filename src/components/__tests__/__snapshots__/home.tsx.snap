// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Array [
  <button
    className="sc-ieebsP iFQabi"
    onClick={[Function]}
    type="button"
  >
    Skip to main content
  </button>,
  <div
    className="sc-fotPbf fETlyZ"
  >
    <header
      className="sc-fFehDp dNyUqG"
    >
      <div
        className="sc-jRQAMF jXwQMc"
      >
        <img
          alt=""
          className="sc-furvIG glFGhe"
          src="https://github.com/fed.png"
        />
        <h1
          className="sc-gKckTs dcpsJO"
        >
          Federico Kn√ºssel
        </h1>
        <p
          className="sc-iCfLBT iujKDL"
        >
          I'm a software engineer with a keen interest in web development, and I help teams ship clean and maintainable code.
           Feel free to hit me up on 
          <a
            className="sc-pVTma OsAqn"
            href="obfuscated"
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "direction": "rtl",
                "unicodeBidi": "bidi-override",
              }
            }
          >
            moc.liamg@lessunkf
          </a>
          .
        </p>
      </div>
      <nav
        aria-label="Social media links"
        className="sc-ksdxAp dIQDEe"
      >
        <a
          className="sc-jrQzUz sc-kDThTU joXqZV hPRemR sc-hBURRC gsevGF"
          href="https://twitter.com/fknussel"
          rel="noopener noreferrer"
          target="_blank"
        >
          Twitter
        </a>
        <a
          className="sc-jrQzUz sc-iqsfdx joXqZV gTwbGu sc-hBURRC gsevGF"
          href="https://github.com/fed"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <a
          className="sc-jrQzUz sc-crHlIS joXqZV hupHmo sc-hBURRC gsevGF"
          href="https://medium.com/@fknussel"
          rel="noopener noreferrer"
          target="_blank"
        >
          Medium
        </a>
        <a
          className="sc-jrQzUz sc-egiSv joXqZV cPeTIl sc-hBURRC gsevGF"
          href="https://codepen.io/fede"
          rel="noopener noreferrer"
          target="_blank"
        >
          CodePen
        </a>
        <a
          className="sc-jrQzUz sc-bqiQRQ joXqZV cWrlVm sc-hBURRC gsevGF"
          href="https://linkedin.com/in/fknussel"
          rel="noopener noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
      </nav>
    </header>
    <main
      aria-label="Blog posts"
      className="sc-bkkfTU kpoTKc"
    >
      <p>
        Content
      </p>
    </main>
  </div>,
  <footer
    aria-label="Social media links"
    className="sc-bdvvaa dpGNEF"
  >
    <ul
      className="sc-gsDJrp YdNwt"
    >
      <li
        className="sc-dkPtyc gAblvE"
      >
        <a
          className="sc-hKwCoD iKfAMc"
          href="https://twitter.com/fknussel"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Twitter"
            className="sc-eCImvq kiUxIk"
            src="test-file-stub"
          />
        </a>
      </li>
      <li
        className="sc-dkPtyc gAblvE"
      >
        <a
          className="sc-hKwCoD iKfAMc"
          href="https://github.com/fed"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub"
            className="sc-eCImvq kiUxIk"
            src="test-file-stub"
          />
        </a>
      </li>
      <li
        className="sc-dkPtyc gAblvE"
      >
        <a
          className="sc-hKwCoD iKfAMc"
          href="https://medium.com/@fknussel"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Medium"
            className="sc-eCImvq kiUxIk"
            src="test-file-stub"
          />
        </a>
      </li>
      <li
        className="sc-dkPtyc gAblvE"
      >
        <a
          className="sc-hKwCoD iKfAMc"
          href="https://codepen.io/fede"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="CodePen"
            className="sc-eCImvq kiUxIk"
            src="test-file-stub"
          />
        </a>
      </li>
      <li
        className="sc-dkPtyc gAblvE"
      >
        <a
          className="sc-hKwCoD iKfAMc"
          href="https://linkedin.com/in/fknussel"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="LinkedIn"
            className="sc-eCImvq kiUxIk"
            src="test-file-stub"
          />
        </a>
      </li>
    </ul>
  </footer>,
]
`;
